### DISCUSSION TITLE
Why VCSIM takes 3 hours to create 4000 vms setup?

### DISCUSSION THREAD

**Discussion by SurendraKadamArista** [vmware/govmomi#3304] [Community Help]
we bring up vcsim with 0 inventory and then using govc commands we bring up complete 4k vms setup having 400 hosts, 4 dvs and 4000 vms but this govc script takes 3 hrs to create this whole setup up. Any idea why its taking this much time to create such setup?

**Comment by kbrock** [comment]
Could you share the vcsim command you used on the command line? (or passing to your docker container?)

When I was setting this up, I was finding that the number of vms got quite large quickly depending upon how the parameters I passed.

ASIDE: I agree that creating tens of thousands of vms was quite slow.

**Comment by dougm** [comment]
Using the v0.36.0 release of vcsim on Linux:
```console
% time vcsim -standalone-host 0 -host 400 -vm 4000 -stdinexit >/dev/null </dev/null

real	3m7.009s
user	0m20.785s
sys	0m2.893s
```
Note: using `-stdinexit` + `</dev/null` causes vcsim to exit after the inventory is created.
I'm sure we could make some optimizations to improve, but ~3 minutes vs ~3 hours.. more details would useful. And/or output of the command above.
